nixos/modules/services/misc/mpdscribble.nix

```diff
@@ -0,0 +1,292 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mpdscribble;
+
+  mkSection = cfg: optname: secname:
```

Comment:
Options for each backend could be also reduced using a generic function, like you did for the configuration file.